### PR TITLE
Test DATE support in JDBC connectors

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java [hunk 1]
```diff
@@ -204,6 +206,30 @@ public void testDecimalExceedingPrecisionMax()
         testUnsupportedDataType("decimal(50,0)");
     }
 
+    @Test
+    public void testDate()
+    {
+        LocalDate northernHemisphereWinter = LocalDate.of(2017, 1, 1);
+        LocalDate northernHemisphereSummer = LocalDate.of(2017, 7, 1);
+        // America/Sao_Paulo is interesting because it has DST changes at midnight. However, for this values to be useful, tests need to be run with JVM in that time zone.
```
Comment (findepi) [inline presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java]: While Kathmandu is really cool zone, it doesn't have DST and never had....

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/StandardReadMappings.java [hunk 2]
```diff
@@ -126,9 +128,9 @@ public static ReadMapping dateReadMapping()
         return longReadMapping(DATE, (resultSet, columnIndex) -> {
             // JDBC returns a date using a timestamp at midnight in the JVM timezone
             long localMillis = resultSet.getDate(columnIndex).getTime();
-            // Convert it to a midnight in UTC
+            // Convert it to a ~midnight in UTC. This won't be precisely midnight if JVM zone had a 'forward' DST change on that particular date, exactly at midnight.
             long utcMillis = ISOChronology.getInstance().getZone().getMillisKeepLocal(UTC, localMillis);
-            // convert to days
+            // convert to days (this also truncates excess millis if utcMillis isn't exactly midnight)
             return MILLISECONDS.toDays(utcMillis);
```
Comment (findepi) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/StandardReadMappings.java]: Only this computation could be replaced with very simple and neat:
```
return resultSet.getObject(columnIndex, LocalDate.class).toEpochDays();
```

except that it doesn't work for redshift (old postgres driver) and sql server (even if we upgrade to newest driver).

File: presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java (hunk 1 above)
Comment (losipiuk) [reply presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java]: I saw the change to `pom.xml` in some other PR. Why isn't it here?

File: presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java [hunk 3]
```diff
@@ -204,6 +206,30 @@ public void testDecimalExceedingPrecisionMax()
...
+        LocalDate dstChangeBackwardInSaoPaulo = LocalDate.of(2017, 2, 19);
+        LocalDate dstChangeForwardInSaoPaulo = LocalDate.of(2017, 10, 15);
+
+        DataTypeTest.create()
```
Comment (losipiuk) [inline presto-mysql/src/test/java/com/facebook/presto/plugin/mysql/TestMySqlTypeMapping.java]: We want to test it in different session TZs - right?

**Comment by findepi** [comment]
I will post this PR anew later.

**Pull request closed without merging**